### PR TITLE
Custom Post Type Templates

### PR DESCRIPTION
**Pull request by karabishopgreenpeace**
# Pull Request Overview

See https://www.notion.so/greenpeaceaustraliapacific/P4-Page-Type-Templates-df543163a9b04494b96de51a116e2b19?pvs=4

To include a custom banner on custom post types. We need to include theme support for Template Parts. Also a custom post type template.

I have updated functions.php to support template parts;
Screenshot 2024-01-30 at 1 50 29 pm

Added a custom post type (single-news.twig)

Added in single.twig to check if custom post-type = News, Then use single-news.twig
Screenshot 2024-01-30 at 1 53 11 pm

In single-news.twig, show banner-news, footer-news and archive template parts.

Screenshot 2024-01-30 at 1 55 23 pm

# 📷 Screenshots

## 🏁 PR Checklist

Reviewer checklist:
- [ ] ✨ This PR includes new user-facing feature and/or settings
- [ ] 📝 Documentation for new features/settings are in PR description (to go to wiki or Coral docs) 

Contributor checklist:
- [ ] Reference a GitHub issue, Jira ticket, or Notion card in the description 💯
- [ ] 📝 Update the CHANGELOG.md (Put under `[Unreleased]`). Include link to this PR.
- [ ] 📝 Add JSDoc code documentation
- [ ] 🏷 Add flow types. (If needed.)
- [ ] ✅ Add unit tests (`*.test.js`) and/or Cypress Tests.
- [ ] 📖 Create Storybook stories. (For any applicable Components.)
- [ ] 📸 Update snapshots.  Run `yarn test:watch`, press 'a' for "all tests", and "u" to update snaps if needed.
- [ ] Run tests (`yarn test` and `yarn flow`) and clean up any errors 🚀
- [ ] Select 1-2 Reviewers
- [x] Feel good about yourself, you're doing great! 🥳

### CODE REVIEW

**Review by ShoryuDZ — COMMENTED** [review]
Relatively happy with this, but some caveats:
1. Due to the complexity in implementation, I think you're going to have to write some pretty comprehensive docs on how all of this works. Include exact slug names because I can see that becoming a bit of a stumbling block for people.
2. With the parts folder, if we could add a comment to each file highlighting what its for, and where to find the content, that would reduce confusion

File: parts/archive.html [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<!-- wp:group {"backgroundColor":"tertiary"} -->
```
[the remark this answers is not in the document]
Comment (karabishopgreenpeace) [reply parts/archive.html:1]: I'm not sure what you mean...

[the remark this answers is not in the document]
Comment (karabishopgreenpeace) [reply parts/archive.html:1]: Ah. Right. You mean in the HTML file? I've added a comment where they go and where they will be displayed.

File: templates/single.twig [hunk 2]
```diff
@@ -5,6 +5,14 @@
     <!-- Custom Post Types -->
     {% if ( post.page_types.0.slug == 'news' )%}
         {% include 'single-news.twig' %}
+    {% elseif ( post.page_types.0.slug == 'team-leadership' )%}
+        {% include 'single-team-leadership.twig' %}
+    {% elseif ( post.page_types.0.slug == 'research-and-reports' )%}
+        {% include 'single-report.twig' %}
+    {% elseif ( post.page_types.0.slug == 'green-guardian-article' )%}
+        {% include 'single-gg-article.twig' %}
+    {% elseif ( post.page_types.0.slug == 'green-guardian-testimonial' )%}
```
[the remark this answers is not in the document]
Comment (karabishopgreenpeace) [reply templates/single.twig:14]: The slugs are based on the Page Type, so all posts with that Page Type will have that slug. They're not arbitrary slugs.

File: parts/archive.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (ShoryuDZ) [reply parts/archive.html:1]: Yeah for each file in `/parts` if we could remove the commented out code which isn't being used and replace it with a short description of where to find/edit the content for said templates that would be great, happy to walk through it with you if its unclear.

**Review by ShoryuDZ — APPROVED** [review]
LGTM, I made some minor changes @karabishopgreenpeace just with code styling/merge conflicts but I'll merge this now :)

**Pull request merged**